### ISSUE TITLE
Retrieve an access object's membership / ancestry

### ISSUE THREAD

**Issue by SaschaAdler** [crkn-rcdr/Access-Platform#94]
A manifest of collection can be a member of one or more collections.

An accessor method is required for retrieving the collection(s) a given object is a member of. This method can be used recursively to determine the full ancestry of an access object, i.e. figuring out the membership of the collections the object is a member of, etc. etc.

It might be worth memoizing the recursive lookup because the further back you go, the more likely it is you're looking up the same collections over and over.

**Comment by SaschaAdler** [comment]
Depends on #93.

**Comment by SaschaAdler** [comment]
Two AccessHandler methods here:

```ts
async getMembership(id: Noid): Promise<{
  id: Noid;
  slug: Slug;
  label: Text;
}[]>
```

1. Either rename `isMemberOf` (which feels like a weird name to me) to this, or use `isMemberOf` here. We can probably set `isMemberOf` to `private`.
2. After performing the view lookup, use `find` to get the Collections' slugs and labels. `find` might have a hard time with `label` because of some typing weirdness I have to sort out in a separate issue. If so, don't worry about incorporating `label` yet.
3. Return an array of objects typed as above.

```ts
async getDescent(id: Noid): Promise<{
  id: Noid;
  slug: Slug;
  label: Text;
}[][]>
```

Note that this is an array of arrays; we also should refactor the return types like so:

```ts
type CollectionListing = {
  id: Noid;
  slug: Slug;
  label: Text;
};

//...

async getMembership(id: Noid): Promise<CollectionListing[]>;
async getDescent(id: Noid): Promise<CollectionListing[][]>;
```

We're building breadcrumbs. Effectively, if `alice` is a member of `bob`, `bob` is a member of `charlie` and `debbie`, and `charlie` and `debbie` aren't members of anything, return something like

```ts
[
  ["alice", "bob", "charlie"],
  ["alice", "bob", "debbie"]
]
```

except with `CollectionListing` objects instead of strings. We'll need to use `getMembership` on every id we see, and we might want to run this recursively?
